wp-content/civi-extensions/goonjcustom/Civi/DroppingCenterService.php

```diff
@@ -144,4 +295,21 @@ private static function isViewingDroppingCenter($tabsetName, $context) {
     return (int) $entitySubtypeValue === $subtypeId;
   }
 
+  /**
+   *
+   */
+  public static function getSubtypeNameByEntityId($entityID) {
+    $getSubtypeName = civicrm_api4('Eck_Collection_Camp', 'get', [
+      'select' => [
+        'subtype:name',
+      ],
+      'where' => [
+            ['id', '=', $entityID],
+      ],
+    ]);
+    $entityData = $getSubtypeName[0] ?? [];
+    $subtypeName = $entityData['subtype:name'] ?? NULL;
+    return $subtypeName;
+  }
+
```

Comment:
Since we get the `entityID` in the s`etGoonOffice` function, we want to include some checks there. The same logic will apply to the collection camp, ensuring that the `setGoonOffice` function only runs for specific subtypes. Please let me know if you have any questions regarding this.